### PR TITLE
Fix dependencies for 4.3.1

### PR DESCRIPTION
**Pull request by mwcraig**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mwcraig** [comment]
@conda-forge-admin please rerender

**Comment by mwcraig** [comment]
@conda-forge-admin, rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,14 +38,13 @@ requirements:
     - cython
     - setuptools_scm
     - extension-helpers
-    - numpy 1.18.*  # [py<39 and build_platform == target_platform and python_impl == 'cpython']
+    - numpy 1.17.*  # [py<39 and build_platform == target_platform and python_impl == 'cpython']
     - numpy         # [py>=39 or build_platform != target_platform or  python_impl == 'pypy']
```
Comment (isuruf) [inline recipe/meta.yaml:42]:
```suggestion
    - numpy
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -38,14 +38,13 @@ requirements:
...
   run:
     - python
     - {{ pin_compatible("numpy") }}
-    - numpy >=1.18
-    - pyerfa >=2.0
-    - pyyaml >=3.13
+    - numpy >=1.17
```
Comment (isuruf) [inline recipe/meta.yaml:46]:
```suggestion
```

Comment (mwcraig) [reply recipe/meta.yaml:46]: Thanks @isuruf -- I'm confused though, where does the minimum numpy version get enforced at a package level if not here? I understand that conda-forge sets minimum numpy versions for each version of python already, I just always seem to get this wrong...

Comment (isuruf) [reply recipe/meta.yaml:46]: See https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by mwcraig** [comment]
@conda-forge-admin, rerender

**Comment by mwcraig** [comment]
Looks like still no numpy 1.17...

**Comment by isuruf** [comment]
We dropped support for 1.17 last month

**Comment by mwcraig** [comment]
Offline discussion led to: Don't specify minimum numpy, let it default to what conda-forge supports/builds.

**Pull request merged**